pkg/chunk/util/util.go

```diff
@@ -125,3 +127,22 @@ func EnsureDirectory(dir string) error {
 	}
 	return err
 }
+
+// ReadCloserWithContextCancelFunc helps with cancelling the context when closing a ReadCloser.
+// Note: The consumer of ReadCloserWithContextCancelFunc should always call the Close method when it is done reading which otherwise could cause a resource leak.
```

Comment:
This comment also belongs on the `GetObject` method of the `ObjectClient` interface, without referencing `ReadCloserWithContextCancelFunc`. Users of `ObjectClient` don't need to know internal implementation details.